app/models/clubhouse.rb

```diff
@@ -0,0 +1,36 @@
+class Clubhouse < ApplicationRecord
+  has_many :posts, dependent: :destroy
+  has_many :memberships, dependent: :destroy
+
+  validates :title, presence: true
+
+  def self.with_owner(owner, properties)
+    clubhouse = Clubhouse.create(properties)
+    membership = Membership.new(
+      user: owner,
+      clubhouse:,
+      role: 'owner'
+    )
+    return clubhouse if membership.save
+
+    nil
+  end
+
+  def owners
+    users_of_role('owner')
+  end
+
+  def admins
+    users_of_role('admin')
+  end
+
+  def members
+    users_of_role('member')
+  end
+
+  private
+
+  def users_of_role(role)
+    Membership.where('clubhouse_id = ? AND role = ?', id, role).map(&:user)
+  end
```

Comment:
There is an interesting problem here, which often becomes a pitfall in these apps.
Since you are using `map` at the end of the query, this method could result in a performance hit as data volume increases. `map` is an array method, not an ActiveRecord/SQL command. This means that, in order to use `map`, Rails has to parse the ActiveRecord query result to an array object and store that array in memory. Having thousands or millions of membership objects (and user objects after the `map` call) stored in memory during runtime can kill your app resources.

This problem happens every time you mix array methods with ActiveRecord/SQL call chains, and it feels intuitive because ActiveRecord instances behave pretty much like arrays, but they are not the same and you must keep this in mind.
 
To keep this performant it's necessary to always try to build the whole query with ActiveRecord calls, and forget about array calls unless you consider that data scaling is not an issue for that scenario

Reply:
Another alternative
```ruby
def users_of_role(role)
  role_memberships =  memberships.where(role: role)
  users.where(id: role_memberships.select(:user_id))
end
```

This would consist of 2 queries, but they are more scalable than using array calls